packages/components/src/scroll-lock/test/index.js

```diff
@@ -41,6 +41,14 @@ describe( 'scroll-lock', () => {
 		wrapper = mount( <ScrollLock /> );
 		expectLocked( true );
 		wrapper.unmount();
-		expectLocked( false );
+
+		// Running cleanup functions now works asynchronously. the unofficial
+		// enzyme adapter for react 17 we're currently using does not account
+		// for this, yet. So for now, we'll use setTimeout to wait for cleanup.
+		//
+		// @see https://reactjs.org/blog/2020/08/10/react-v17-rc.html#effect-cleanup-timing
+		setTimeout( () => {
```

Comment:
Unit tests have fake timers enabled so it might be a coincidence that `setTimeout` works as expected here. More details on the topic:

https://jestjs.io/docs/timer-mocks#advance-timers-by-time

I haven't tested it but it's probably more future-proof to code it as:

```js
jest.advanceTimersByTime( 1 ); // not sure about the time value
expectLocked( false );
```

Reply:
cc @vdwijngaert since I see that change introduced in [this commit](https://github.com/WordPress/gutenberg/commit/79b51c9d52e209fd5e51c8480be8962b5596378f) in the [React v17 upgrade PR](https://github.com/WordPress/gutenberg/pull/26847).

Reply:
I don't remember this test failing even before I merged in the React 17 upgrade branch, maybe it's because I had installed a more recent version of `@wojtekmaj/enzyme-adapter-react-17` (0.6.2 instead of 0.3.1). I'll try reverting this commit altogether to see if it works fine.

Reply:
Reverting the commit didn't work :) But using `jest.advanceTimersByTime` instead worked fine. [`3e095a7` (#29118)](https://github.com/WordPress/gutenberg/pull/29118/commits/3e095a70589f5a3dc64a309d6159875bc742ef29)

Reply:
I'm happy but surprised it worked on first try 🤣